File: src/db/models/Task.ts
/**
 * db/models/Task.ts
 * tasks 테이블의 레코드 구조를 정의한다.
 * 사용자가 등록한 원본 업무 요청을 저장한다.
 *
 * 테이블 컬럼:
 *   id          UUID PK
 *   title       VARCHAR
 *   description TEXT
 *   requester_id VARCHAR
 *   deadline    TIMESTAMP (nullable)
 *   status      VARCHAR  (pending | running | completed | failed)
 *   created_at  TIMESTAMP
 *   updated_at  TIMESTAMP
 */

export interface TaskRecord {
  id: string;
  title: string;
  description: string;
  requesterId: string;
  deadline: string | null;
  status: string;
  createdAt: string;
  updatedAt: string;
}

File: app/models/task.py
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel


class TaskStatus(str, Enum):
    pending   = "pending"
    running   = "running"
    completed = "completed"
    failed    = "failed"


class TaskInput(BaseModel):
    title:       str
    description: str
    deadline:    Optional[str] = None


class Task(TaskInput):
    id:        str
    status:    TaskStatus
    createdAt: str
    updatedAt: str
